### PR TITLE
[NEW] website_slides: Backport to V8.

### PR DESCRIPTION
**Pull request by sergio-teruel**
First backport website_slides module to v8.0

### CODE REVIEW

File: website_slides/__init__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import controllers
```
Comment (pedrobaeza) [inline website_slides/__init__.py]: Use relative import

File: website_slides/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+{
+    'name': 'Slides',
+    'version': '1.0',
+    'summary': 'Share and Publish Videos, Presentations and Documents',
+    'category': 'website',
+    'description': """
+Share and Publish Videos, Presentations and Documents'
```
Comment (pedrobaeza) [inline website_slides/__openerp__.py]: Extract it to a README.rst file and use OCA's template.

File: website_slides/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+ * Website Application
+ * Channel Management
+ * Filters and Tagging
+ * Statistics of Presentation
+ * Channel Subscription
+ * Supported document types supported: PDF, images, YouTube videos and
+ Google Drive documents)
+""",
+    'author': 'Odoo SA',
```
Comment (pedrobaeza) [inline website_slides/__openerp__.py]: Add your credentials and OCA's

File: website_slides/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+
+from openerp.addons.web import http
+from openerp.exceptions import AccessError
+from openerp.http import request
+from openerp.tools.translate import _
+
+_logger = logging.getLogger(__name__)
+
+
+class website_slides(http.Controller):
```
Comment (pedrobaeza) [inline website_slides/controllers/main.py]: Use CamelCase for class names

**Comment by pedrobaeza** [comment]
Please check PEP8 errors

**Comment by OSguard** [comment]
where is to original source taken? license information are completely missing!

File: website_slides/README.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+
+Credits
+=======
+
+Contributors
+------------
+
+* Carlos Dauden (Incaser) <carlos@incaser.es>
+* Sergio Teruel (Incaser) <sergio@incaser.es>
```
Comment (gurneyalex) [inline website_slides/README.rst:48]: I'm pretty sure you are not the only contributors to this module.

**Comment by gurneyalex** [comment]
please preserve the original commit history and authors. 

:-1: in the current state of the PR.

**Comment by pedrobaeza** [comment]
Hi, @sergio-incaser, please check how to correctly extract the module with its commit history here: https://github.com/odoomrp/odoomrp-wip/wiki/Migration-to-OCA, so this is the proposed steps:
- Keep a copy of your current branch.
- Over a clean branch of website, followed the described method on the link to extract the module from the master branch where you originally took it.
- Copy your current module implementation over it, and make a new commit for your changes.
- Propose this branch as a PR.

**Comment by sergio-teruel** [comment]
New PR #52.

**Comment by yassineteimi** [comment]
Are this module developpements have been completed ? Thanks.

**Comment by sergio-teruel** [comment]
Yes, of course. This is only a backport from website_slides v9.

**Comment by yassineteimi** [comment]
Thanks sergio, Yes I was following it since the begining, I'll test it right now. Regards.

**Comment by yassineteimi** [comment]
I've the following error when installing the module :

 Odoo Server Error
Traceback (most recent call last):
  File "/opt/odoo/odoo/openerp/http.py", line 537, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo/openerp/http.py", line 574, in dispatch
    result = self._call_function(*_self.params)
  File "/opt/odoo/odoo/openerp/http.py", line 310, in _call_function
    return checked_call(self.db, *args, *_kwargs)
  File "/opt/odoo/odoo/openerp/service/model.py", line 113, in wrapper
    return f(dbname, _args, *_kwargs)
  File "/opt/odoo/odoo/openerp/http.py", line 307, in checked_call
    return self.endpoint(_a, *_kw)
  File "/opt/odoo/odoo/openerp/http.py", line 803, in __call__
    return self.method(_args, *_kw)
  File "/opt/odoo/odoo/openerp/http.py", line 403, in response_wrap
    response = f(_args, *_kw)
  File "/opt/odoo/odoo/addons/web/controllers/main.py", line 948, in call_button
    action = self._call_kw(model, method, args, {})
  File "/opt/odoo/odoo/addons/web/controllers/main.py", line 936, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, _args, *_kwargs)
  File "/opt/odoo/odoo/openerp/api.py", line 241, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/opt/odoo/odoo/openerp/addons/base/module/module.py", line 450, in button_immediate_install
    return self._button_immediate_function(cr, uid, ids, self.button_install, context=context)
  File "/opt/odoo/odoo/openerp/api.py", line 241, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/opt/odoo/odoo/openerp/addons/base/module/module.py", line 498, in _button_immediate_function
    registry = openerp.modules.registry.RegistryManager.new(cr.dbname, update_module=True)
  File "/opt/odoo/odoo/openerp/modules/registry.py", line 370, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/opt/odoo/odoo/openerp/modules/loading.py", line 355, in load_modules
    loaded_modules, update_module)
  File "/opt/odoo/odoo/openerp/modules/loading.py", line 255, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/opt/odoo/odoo/openerp/modules/loading.py", line 176, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/opt/odoo/odoo/openerp/modules/loading.py", line 118, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/opt/odoo/odoo/openerp/tools/convert.py", line 901, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/opt/odoo/odoo/openerp/tools/convert.py", line 987, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/opt/odoo/odoo/openerp/tools/convert.py", line 853, in parse
    self._tags[rec.tag](self.cr, rec, n, mode=mode)
  File "/opt/odoo/odoo/openerp/tools/convert.py", line 693, in _tag_record
    model = self.pool[rec_model]
  File "/opt/odoo/odoo/openerp/modules/registry.py", line 102, in __getitem__
    return self.models[model_name]
ParseError: "mail.template" while parsing /opt/odoo/odoo/addons/website_slides/data/website_slides_data.xml:18, near
<record id="slide_template_published" model="mail.template">
            <field name="name">Slide Published</field>
            <field name="subject">New ${object.slide_type} published on ${object.channel_id.name}</field>
            <field name="model_id" ref="model_slide_slide"/>
            <field name="auto_delete" eval="True"/>
            <field name="body_html">
&lt;div style="padding:0px; margin:0px;"&gt;
&lt;table cellpadding="0" cellspacing="0" style="width: 600px; margin: 10px 0px 0px 0px; vertical-align: top; padding: 0px; font-family:arial; font-size:12px; background-color:#ffffff"&gt;
    &lt;tbody&gt;&lt;tr&gt;
        &lt;td style="width: 600px; text-align:center" valign="center"&gt;
            &lt;a href="${user.company_id.website}" style="text-decoration:none"&gt;
                &lt;img alt="Odoo" src="${ctx['base_url']}/logo.png" style="display:block; border:none; min-height:60px; margin:0 auto;"&gt;
            &lt;/a&gt;
        &lt;/td&gt;
    &lt;/tr&gt;&lt;/tbody&gt;
&lt;/table&gt;
&lt;/div&gt;
&lt;div style="padding:0px; margin:0px;"&gt;
&lt;table cellpadding="0" cellspacing="0" style="width: 600px; margin: 10px 0px 0px 0px; vertical-align: top; padding: 0px; font-family:arial; font-size:12px;"&gt;
    &lt;tbody&gt;&lt;tr&gt;
        &lt;td style="width: 600px; text-align:center; color: #ffffff; background-color:#a24689;" valign="center"&gt;
            &lt;h2 style="margin:0px;"&gt;
                New ${object.slide_type} published on ${object.channel_id.name}
            &lt;/h2&gt;
        &lt;/td&gt;
    &lt;/tr&gt;&lt;tr&gt;
        &lt;td style="width: 600px; vertical-align:top; text-align:center; background-color:#FFFFF; color:#414141"&gt;
            &lt;a href="${object.share_url}" style="color: #a24689;"&gt;
                &lt;img alt="${object.name}" src="${ctx['base_url']}/website/image/slide.slide/${object.id}/image" style="height:auto; width:500px; background-color: #cccccc; margin: 16px;"&gt;
            &lt;/a&gt;
            &lt;p style="font-size:24px; font-weight:bold"&gt;&lt;a href="${object.share_url}"&gt;${object.name}&lt;/a&gt;&lt;/p&gt;
            &lt;p style="color: #ccc;"&gt;Published on ${object.write_date}&lt;/p&gt;
            &lt;p&gt;&lt;a href="${object.share_url}"&gt;Click here to open the ${object.slide_type}&lt;/a&gt;&lt;/p&gt;
        &lt;/td&gt;
    &lt;/tr&gt;&lt;/tbody&gt;
&lt;/table&gt;
&lt;/div&gt;
&lt;div style="padding:0px; margin:0px;"&gt;
&lt;table cellpadding="0" cellspacing="0" style="margin: 10px 0px 0px 0px; vertical-align: top;padding: 0px; font-family:arial; font-size:12px; color: #ffffff; background-color:#8f8f8f;"&gt;
    &lt;tbody&gt;&lt;tr&gt;
        &lt;td style="width: 600px; font-size:12px; text-align:center; padding-top:10px; color:#a24689; padding-bottom:5px"&gt;
            &lt;a href="${object.share_url}" style="color:#a24689"&gt;View in browser&lt;/a&gt; |
            &lt;a href="/page/contactus" style="color:#a24689"&gt;Contact&lt;/a&gt;
        &lt;/td&gt;
    &lt;/tr&gt;&lt;/tbody&gt;
&lt;/table&gt;
&lt;/div&gt;</field>
        </record>

Do you have an idea about this error ? 

Many thanks.

**Pull request closed without merging**